x-pack/plugins/security_solution/server/lib/detection_engine/signals/source_fields_merging/strategies/get_strategy.ts

```diff
@@ -0,0 +1,31 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import { assertUnreachable } from '../../../../../../common';
+import type { ConfigType } from '../../../../../config';
+import { MergeStrategyFunction } from '../types';
+import { mergeAllFieldsWithSource } from './merge_all_fields_with_source';
+import { mergeMissingFieldsWithSource } from './merge_missing_fields_with_source';
+import { mergeNoFields } from './merge_no_fields';
+
+export const getMergeStrategy = (
+  mergeStrategy: ConfigType['alertMergeStrategy']
+): MergeStrategyFunction => {
+  switch (mergeStrategy) {
+    case 'allFields': {
+      return mergeAllFieldsWithSource;
+    }
+    case 'missingFields': {
+      return mergeMissingFieldsWithSource;
+    }
+    case 'noFields': {
+      return mergeNoFields;
+    }
+    default:
+      return assertUnreachable(mergeStrategy);
```

Comment:
++ on the exhaustive switch here. Nice errors coming from the kibana console as well when mis-configured 👍 

![image](https://user-images.githubusercontent.com/2946766/124032263-96f6d980-d9b5-11eb-9697-f06458c02555.png)